File: app/src/main/java/com/example/spring/ListAccountSubscriptionsQueryService.java
package com.example.spring;

import com.example.spring.error.NotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * List subscriptions for an account.
 */
@Service
public class ListAccountSubscriptionsQueryService {

    private final EntitlementsRepository entitlementsRepository;

    public ListAccountSubscriptionsQueryService(EntitlementsRepository entitlementsRepository) {
        this.entitlementsRepository = entitlementsRepository;
    }

    @Transactional(readOnly = true)
    public java.util.List<Subscription> handle(ListAccountSubscriptionsQueryRequest request) {
        // === GEN_REGION_START: service-body ===
        // spec-driven body for Query ListAccountSubscriptions.
        // intent: List subscriptions for an account.
        // reads: EntitlementsRepository.listSubscriptionsByAccount via `subscriptions`.
        return this.entitlementsRepository.listSubscriptionsByAccount(request.accountId());
        // === GEN_REGION_END: service-body ===
    }
}

File: app-python/entitlements_app/services/list_account_subscriptions_query_service.py
"""List subscriptions for an account."""

from decimal import Decimal

from entitlements_app.requests.list_account_subscriptions_query_request import ListAccountSubscriptionsQueryRequest
from entitlements_app.entities import Subscription
from entitlements_app.ports import EntitlementsRepository


class ListAccountSubscriptionsQueryService:
    """Service handling Query `ListAccountSubscriptions`."""

    def __init__(self, entitlements_repository: EntitlementsRepository) -> None:
        self.entitlements_repository: EntitlementsRepository = entitlements_repository

    def handle(self, request: ListAccountSubscriptionsQueryRequest) -> list[Subscription]:
        # spec-driven body for Query ListAccountSubscriptions.
        # intent: List subscriptions for an account.
        # reads: EntitlementsRepository.listSubscriptionsByAccount via `subscriptions`.
        return self.entitlements_repository.listSubscriptionsByAccount(request.accountId)
